app/assets/javascripts/oxalis/model/reducers/flycam_reducer.js

```diff
@@ -191,12 +193,18 @@ function FlycamReducer(state: OxalisState, action: Action): OxalisState {
       return setDirectionReducer(state, action.direction);
     }
 
-    case "MOVE_FLYCAM":
+    case "MOVE_FLYCAM": {
+      let newMatrix = M4x4.translate(action.vector, state.flycam.currentMatrix, []);
+      if (newMatrix.includes(NaN)) {
+        // if there was a miscalculation, do not move the matrix
+        newMatrix = state.flycam.currentMatrix;
```

Comment:
You can also simply early-return the old state here and make `newMatrix`const again:
```suggestion
        return state;
```

Reply:
Or alternatively, revert the changes in this `case` and check the action.vector for NaN in the same way as you do in the `moveReducer` :)

`if (action.vector.includes(NaN)) return state;`